src/compiler/checker.ts

```diff
@@ -13629,8 +13688,12 @@ namespace ts {
             return result;
         }
 
+        function getCrossProductUnionSize(types: readonly Type[]) {

```

Comment:
I don't know if it matters much, but I used the term "cartesian product" in `typeRelatedToDiscriminatedType` when determining discriminant union combinations (see `cartesianProduct` in core.ts). I usually see "cartesian product" used in reference to [set theory](https://en.wikipedia.org/wiki/Cartesian_product) and "cross product" used in reference to [vector algebra](https://en.wikipedia.org/wiki/Cross_product).

Reply:
Yeah, "cartesian product" is probably the mathematically correct term. SQL calls it a "cross join" which may be even better.